velox/common/memory/HashStringAllocator.cpp

```diff
@@ -99,6 +101,41 @@ void HashStringAllocator::clear() {
   for (auto i = 0; i < kNumFreeLists; ++i) {
     new (&free_[i]) CompactDoubleList();
   }
+
+  static const auto kHugePageSize = memory::AllocationTraits::kHugePageSize;
+  for (auto i = 0; i < pool_.numRanges(); ++i) {
+    auto topRange = pool_.rangeAt(i);
+    auto topRangeSize = topRange.size();
+    if (topRangeSize >= kHugePageSize) {
+      VELOX_CHECK_EQ(0, topRangeSize % kHugePageSize);
+    }
+
+    for (int64_t subRangeStart = 0; subRangeStart < topRangeSize;
+         subRangeStart += kHugePageSize) {
+      auto range = folly::Range<char*>(
+          topRange.data() + subRangeStart,
+          std::min<int64_t>(topRangeSize, kHugePageSize));
+      const auto size = range.size() - simd::kPadding;
+      auto* end = reinterpret_cast<Header*>(range.data() + size);
+      auto* header = reinterpret_cast<Header*>(range.data());
```

Comment:
Can we have define macro for this?
```
#define HEADER(ptr) reinterpret_cast<Header*>(ptr) 
```

Reply:
Because Velox does not recommend using macros, so I created a new `static Header* castToHeader(const void* data)` method.